pymc3/sampling.py

```diff
@@ -10,8 +10,40 @@
 
 __all__ = ['sample', 'iter_sample']
 
-
-def sample(draws, step, start=None, trace=None, chain=0, njobs=1, tune=None,
+def assign_step_methods(model, step):
+    
+    steps = []
+    assigned_vars = set()
+    if step is not None:
+        steps = np.append(steps, step).tolist()
+        try:
+            assigned_vars = assigned_vars | set(step.vars)
+        except AttributeError:
+            for s in step.methods:
+                assigned_vars = assigned_vars | set(s.vars)
+    
+    # Use competence classmethods to select step methods for remaining variables
+    selected_steps = {s:[] for s in step_methods.step_method_registry}
+    for var in model.free_RVs:
+        if not var in assigned_vars:
+               
+            competences = {s:s._competence(var) for s in
+                             step_methods.step_method_registry}
+            
+            selected = max(competences.keys(), key=(lambda k: competences[k]))
+        
+            print('Assigned {0} to {1}'.format(selected, var))
+            selected_steps[selected].append(var)
+    
+    # Instantiate all selected step methods
+    steps += [s(vars=selected_steps[s]) for s in selected_steps if selected_steps[s]]
+    
+    if len(steps)==1:
+        steps = steps[0]    
+                
+    return model, steps
```

Comment:
why return model here? seems like we want to be able to say `sample(300, step=assign_step_methods(model, []))`